### ISSUE TITLE
unexpected change  of visual and entering editor mode for question on change of date time filter of a saved question

### ISSUE THREAD

**Issue by chris299** [metabase/metabase#12014]
**Describe the bug**
we experience the following behavior:
a saved question with bar graph visualization
https://...de/question/1?TIME_UNIT=DAY&TIME_FILTER=past30days
changes visualization to table and enters editor mode (used by a read-only user)
when changing the date time filter to
https://....de/question/1?TIME_UNIT=WEEK&TIME_FILTER=2020-03-09~2020-03-13

first we assumed it may be caused by the parameter with ~, but it happens only with certain time periods, however, it can be reproduced with this question and time periods

it happens not only by changing the parameter in the gui but also by directly calling the second URL with their parameters.

the SQL based question is kind of complex, but that should not make any difference. I feel like it depends on the "Extract-week" stuff.

```
select
time_group,
count(*),
sum(SUBTOTAL),
sum(TOTAL)
from
(
select
 (case 
 when {{TIME_UNIT}} = 'WEEK' then CAST(EXTRACT(WEEK FROM CREATED_AT) as varchar) 
 when {{TIME_UNIT}} = 'DAY' then CAST(EXTRACT(DAY FROM CREATED_AT) as varchar) 
 end) as time_group,
 SUBTOTAL,
 TOTAL
 from ORDERS where 1=1 [[and {{TIME_FILTER}}]]
) 
group by time_group
```

**Logs**
the JS Console Log of Chrome shows one additional entry for the "defective" URL:

vendor.bundle.js?a10c9974cc03fead96be:1 Deprecation warning: value provided is not in a recognized RFC2822 or ISO format. moment construction falls back to js Date(), which is not reliable across all browsers and versions. Non RFC2822/ISO date formats are discouraged and will be removed in an upcoming major release. Please refer to http://momentjs.com/guides/#/warnings/js-date/ for more info.
Arguments: 
[0] _isAMomentObject: true, _isUTC: false, _useUTC: false, _l: undefined, _i: day, _f: undefined, _strict: undefined, _locale: [object Object]
Error
    at Function.createFromInputFallback (https://app-metabase-das-prd.[some domain].de/app/dist/vendor.bundle.js?a10c9974cc03fead96be:1:21278)
    at kt (https://app-metabase-das-prd.[some domain].de/app/dist/vendor.bundle.js?a10c9974cc03fead96be:1:38999)
    at Tt (https://app-metabase-das-prd.[some domain].de/app/dist/vendor.bundle.js?a10c9974cc03fead96be:1:39710)
    at Et (https://app-metabase-das-prd.[some domain].de/app/dist/vendor.bundle.js?a10c9974cc03fead96be:1:39792)
    at i (https://app-metabase-das-prd.[some domain].de/app/dist/vendor.bundle.js?a10c9974cc03fead96be:1:18119)
    at Y (https://app-metabase-das-prd.[some domain].de/app/dist/app-main.bundle.js?a10c9974cc03fead96be:5:437453)
    at https://app-metabase-das-prd.[some domain].de/app/dist/app-main.bundle.js?a10c9974cc03fead96be:5:437756
    at Array.map (<anonymous>)
    at X (https://app-metabase-das-prd.[some domain].de/app/dist/app-main.bundle.js?a10c9974cc03fead96be:5:437730)
    at Object.init (https://app-metabase-das-prd.[some domain].de/app/dist/app-main.bundle.js?a10c9974cc03fead96be:5:439121)

the metabase log does not indicate any problem:
[5f3c89c1-4812-4d1d-8ca5-48affb67bd78] 2020-02-28T20:31:10+01:00 DEBUG metabase.middleware.log GET /api/user/current 200 4.2 ms (3 DB calls) Jetty threads: 4/50 (3 idle, 0 queued) (189 total active threads) Queries in flight: 0
[5f3c89c1-4812-4d1d-8ca5-48affb67bd78] 2020-02-28T20:31:10+01:00 DEBUG metabase.middleware.log GET /api/session/properties 200 3.7 ms (1 DB calls) Jetty threads: 3/50 (3 idle, 0 queued) (190 total active threads) Queries in flight: 0
[5f3c89c1-4812-4d1d-8ca5-48affb67bd78] 2020-02-28T20:31:10+01:00 DEBUG metabase.middleware.log GET /api/database 200 9.8 ms (3 DB calls) Jetty threads: 3/50 (4 idle, 0 queued) (190 total active threads) Queries in flight: 0
[5f3c89c1-4812-4d1d-8ca5-48affb67bd78] 2020-02-28T20:31:10+01:00 DEBUG metabase.middleware.log GET /api/setting 200 472.4 µs (0 DB calls) Jetty threads: 6/50 (1 idle, 0 queued) (190 total active threads) Queries in flight: 0
[5f3c89c1-4812-4d1d-8ca5-48affb67bd78] 2020-02-28T20:31:10+01:00 DEBUG metabase.middleware.log GET /api/card/205 200 13.1 ms (6 DB calls) Jetty threads: 5/50 (1 idle, 0 queued) (191 total active threads) Queries in flight: 0
[5f3c89c1-4812-4d1d-8ca5-48affb67bd78] 2020-02-28T20:31:10+01:00 DEBUG metabase.middleware.log GET /api/alert/question/205 200 1.9 ms (1 DB calls) Jetty threads: 4/50 (3 idle, 0 queued) (192 total active threads) Queries in flight: 0
[5f3c89c1-4812-4d1d-8ca5-48affb67bd78] 2020-02-28T20:31:11+01:00 DEBUG metabase.middleware.log GET /api/database 200 446.8 ms (16 DB calls) Jetty threads: 3/50 (3 idle, 0 queued) (192 total active threads) Queries in flight: 0
[5f3c89c1-4812-4d1d-8ca5-48affb67bd78] 2020-02-28T20:31:11+01:00 DEBUG metabase.middleware.log GET /api/collection/41 200 7.1 ms (3 DB calls) Jetty threads: 4/50 (3 idle, 0 queued) (192 total active threads) Queries in flight: 0
[5f3c89c1-4812-4d1d-8ca5-48affb67bd78] 2020-02-28T20:31:11+01:00 DEBUG metabase.middleware.log GET /api/field/177903 200 11.6 ms (7 DB calls) Jetty threads: 7/50 (1 idle, 0 queued) (193 total active threads) Queries in flight: 0
[5f3c89c1-4812-4d1d-8ca5-48affb67bd78] 2020-02-28T20:31:11+01:00 DEBUG metabase.middleware.log GET /api/field/177903/values 200 5.9 ms (3 DB calls) Jetty threads: 7/50 (1 idle, 0 queued) (193 total active threads) Queries in flight: 0
[5f3c89c1-4812-4d1d-8ca5-48affb67bd78] 2020-02-28T20:31:11+01:00 DEBUG metabase.middleware.log GET /api/field/177942/values 200 8.0 ms (4 DB calls) Jetty threads: 5/50 (2 idle, 0 queued) (193 total active threads) Queries in flight: 0
[5f3c89c1-4812-4d1d-8ca5-48affb67bd78] 2020-02-28T20:31:11+01:00 DEBUG metabase.middleware.log GET /api/field/177942 200 15.7 ms (6 DB calls) Jetty threads: 4/50 (3 idle, 0 queued) (193 total active threads) Queries in flight: 0
[5f3c89c1-4812-4d1d-8ca5-48affb67bd78] 2020-02-28T20:31:11+01:00 DEBUG metabase.middleware.log GET /api/field/177921 200 10.2 ms (6 DB calls) Jetty threads: 4/50 (3 idle, 0 queued) (193 total active threads) Queries in flight: 0
[5f3c89c1-4812-4d1d-8ca5-48affb67bd78] 2020-02-28T20:31:11+01:00 DEBUG metabase.middleware.log GET /api/field/177921/values 200 7.7 ms (4 DB calls) Jetty threads: 3/50 (5 idle, 0 queued) (193 total active threads) Queries in flight: 0
[5f3c89c1-4812-4d1d-8ca5-48affb67bd78] 2020-02-28T20:31:11+01:00 DEBUG metabase.middleware.log GET /api/field/177923 200 9.6 ms (6 DB calls) Jetty threads: 5/50 (2 idle, 0 queued) (193 total active threads) Queries in flight: 0
[5f3c89c1-4812-4d1d-8ca5-48affb67bd78] 2020-02-28T20:31:11+01:00 DEBUG metabase.middleware.log GET /api/field/177923/values 200 5.8 ms (4 DB calls) Jetty threads: 7/50 (1 idle, 0 queued) (193 total active threads) Queries in flight: 0
[5f3c89c1-4812-4d1d-8ca5-48affb67bd78] 2020-02-28T20:31:11+01:00 DEBUG metabase.middleware.log GET /api/field/177927 200 8.9 ms (6 DB calls) Jetty threads: 5/50 (2 idle, 0 queued) (193 total active threads) Queries in flight: 0
[5f3c89c1-4812-4d1d-8ca5-48affb67bd78] 2020-02-28T20:31:11+01:00 DEBUG metabase.middleware.log GET /api/field/177927/values 200 6.5 ms (4 DB calls) Jetty threads: 4/50 (3 idle, 0 queued) (193 total active threads) Queries in flight: 0
[5f3c89c1-4812-4d1d-8ca5-48affb67bd78] 2020-02-28T20:31:11+01:00 DEBUG metabase.middleware.log GET /api/field/138150 200 12.5 ms (7 DB calls) Jetty threads: 3/50 (4 idle, 0 queued) (193 total active threads) Queries in flight: 0
[5f3c89c1-4812-4d1d-8ca5-48affb67bd78] 2020-02-28T20:31:11+01:00 DEBUG metabase.middleware.log GET /api/field/138150/values 200 6.0 ms (4 DB calls) Jetty threads: 3/50 (5 idle, 0 queued) (193 total active threads) Queries in flight: 0
[5f3c89c1-4812-4d1d-8ca5-48affb67bd78] 2020-02-28T20:31:11+01:00 DEBUG metabase.middleware.log GET /api/field/177931 200 10.9 ms (6 DB calls) Jetty threads: 6/50 (1 idle, 0 queued) (193 total active threads) Queries in flight: 0
[5f3c89c1-4812-4d1d-8ca5-48affb67bd78] 2020-02-28T20:31:11+01:00 DEBUG metabase.middleware.log GET /api/field/177955 200 20.7 ms (6 DB calls) Jetty threads: 5/50 (2 idle, 0 queued) (193 total active threads) Queries in flight: 0
[5f3c89c1-4812-4d1d-8ca5-48affb67bd78] 2020-02-28T20:31:11+01:00 DEBUG metabase.middleware.log GET /api/field/177931/values 200 26.6 ms (4 DB calls) Jetty threads: 5/50 (3 idle, 0 queued) (193 total active threads) Queries in flight: 0
[5f3c89c1-4812-4d1d-8ca5-48affb67bd78] 2020-02-28T20:31:11+01:00 DEBUG metabase.middleware.log GET /api/field/177955/values 200 29.8 ms (3 DB calls) Jetty threads: 5/50 (3 idle, 0 queued) (193 total active threads) Queries in flight: 0
[5f3c89c1-4812-4d1d-8ca5-48affb67bd78] 2020-02-28T20:31:11+01:00 INFO metabase.api.card Question's average execution duration is 226524 ms; using 'magic' TTL of 4530 seconds 💾
[5f3c89c1-4812-4d1d-8ca5-48affb67bd78] 2020-02-28T20:32:20+01:00 INFO metabase.query-processor.middleware.cache Caching results for next time for query 💾
[5f3c89c1-4812-4d1d-8ca5-48affb67bd78] 2020-02-28T20:32:20+01:00 INFO metabase.query-processor.middleware.cache Query took 69049 ms to run; miminum for cache eligibility is 5000 ms
[5f3c89c1-4812-4d1d-8ca5-48affb67bd78] 2020-02-28T20:32:20+01:00 DEBUG metabase.middleware.log POST /api/card/205/query 202 [ASYNC: completed] 1.2 mins (29 DB calls) Jetty threads: 2/50 (4 idle, 0 queued) (189 total active threads) Queries in flight: 0
[5f3c89c1-4812-4d1d-8ca5-48affb67bd78] 2020-02-28T20:34:08+01:00 DEBUG metabase.middleware.log POST /api/card/205/query 202 [ASYNC: completed] 6.9 s (27 DB calls) Jetty threads: 2/50 (4 idle, 0 queued) (189 total active threads) Queries in flight: 0

**To Reproduce**
1. save native question with above sql (for sample dataset)
2. configure variables 
time unit=text
time filter= field filter on orders.created_at with widget = datefilter
3. change visual to bar chart (stacked bar to 100%) 
time_group on x, all others on y axis
4. save question
5. call URL with first parameter set (TIME_UNIT=DAY&TIME_FILTER=past30days)
6. call URL with second parameter set (TIME_UNIT=WEEK&TIME_FILTER=2020-03-09~2020-03-13)
->See "save" option on upper right corner and result table instead of bar graph

**Expected behavior**
changing a datetime filter of a question should not lead to entering the editor mode or changing the visual, especially not for read-only-users

**Severity**
for us the problem is a show-stopper for rolling out metabase to normal users in our organisation

**Additional context**
-

**Metabase Diagnostic Info**

```json
{
  "browser-info": {
    "language": "de-DE",
    "platform": "Win32",
    "userAgent": "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/80.0.3987.106 Safari/537.36",
    "vendor": "Google Inc."
  },
  "system-info": {
    "java.runtime.name": "OpenJDK Runtime Environment",
    "java.runtime.version": "11.0.5+10",
    "java.vendor": "AdoptOpenJDK",
    "java.vendor.url": "https://adoptopenjdk.net/",
    "java.version": "11.0.5",
    "java.vm.name": "OpenJDK 64-Bit Server VM",
    "java.vm.version": "11.0.5+10",
    "os.name": "Linux",
    "os.version": "3.10.0-1062.12.1.el7.x86_64",
    "user.language": "en",
    "user.timezone": "GMT"
  },
  "metabase-info": {
    "databases": [
      "h2",
      "sqlserver",
      "postgres",
      "oracle",
      "mysql",
      "mongo"
    ],
    "hosting-env": "unknown",
    "application-database": "mysql",
    "application-database-details": {
      "database": {
        "name": "MariaDB",
        "version": "10.1.29-MariaDB"
      },
      "jdbc-driver": {
        "name": "MariaDB Connector/J",
        "version": "2.5.1"
      }
    },
    "run-mode": "prod",
    "version": {
      "date": "2020-02-25",
      "tag": "v0.34.3",
      "branch": "release-0.34.x",
      "hash": "1a83edb"
    },
    "settings": {
      "report-timezone": "Europe/Berlin"
    }
  }
}
```

**Comment by flamber** [comment]
Hi @chris299 
It sounds like a duplicate of #11964, but I got a little confused, when you said it "enters editor mode", since you say it's a SQL (Native) question, but do you mean it enables the Save button?

**Comment by chris299** [comment]
yes. I meant the enabled save button.
However I think my problem is different to #11964 because I don't get the "new question" headline. also the URL doesn't alter to this long string based editor URL.
so I feel it it is more like #11225 (as we heard stuff like this from our users as well) or #11994
but we see the additional changing in the visualization

**Comment by flamber** [comment]
@chris299 It would help a lot if you could provide steps to reproduce with Sample Dataset. That's the reason why #11225 was closed - and because there wasn't any reply.

**Comment by chris299** [comment]
@flamber I'll try to reproduce the problem in a sample Dataset. Do you have a guideline how to provide this to the community?

**Comment by flamber** [comment]
@chris299 
You should have a database called "Sample Dataset" (since I can see you have a H2 data source).
Then use that for all the questions and dashboards you create, and write down the steps, so I can follow them and reproduce the problem. It's difficult to give exact pointers, since every issue is slightly different, but have a look at https://github.com/metabase/metabase/issues/11964#issuecomment-591352424 and https://www.metabase.com/docs/latest/troubleshooting-guide/bugs.html

**Comment by chris299** [comment]
@flamber we were able to reproduce the problem with the sample data set. I'll change the description accordingly

**Comment by flamber** [comment]
@chris299 With your update, I can see that you're probably seeing two issues (maybe more).
With you steps to reproduce, you're only returning 1 row, which is why it's switching to table visualization because of #11675 - and then it feels like you're seeing #11994 on top of that (or something similar), since it's not starting a full "New question" like #11964.
But I have a feeling that this is combination of multiple things that is causing you problems.
I'll leave this open, since it now has a way to reproduce the problem.

**Comment by chris299** [comment]
I agree on the similarity to #11994 but I don't think it is exactly like #11675 because we saw the problem also with Version 0.33 and we see variations of our sample question with only one row result which do work (they give one bar in the chart)

**Comment by flamber** [comment]
@chris299 Okay, if you saw the problem with 0.33.x, then you should have reported it back then, so we could have possibly fixed it and written a test too.
I've just tested with you steps-to-reproduce and cannot reproduce on 0.33.7.3
Unless I'm misunderstanding something, then this is #11675 combined with (most likely) #11994.

**Comment by chris299** [comment]
we tried to update first and check, if the problem still exists.... but we are fine with the relationship to the other two issues.

**Comment by flamber** [comment]
Fixed by #12476
